src/types-and-traits.rst

```diff
@@ -2101,12 +2101,12 @@ Type Inference
 :t:`[expression]s` and :t:`[pattern]s` within a :t:`type inference root`.
 
 :dp:`fls_ybvrhh96fc7y`
-A :t:`type inference root` is an :t:`expression` whose inner :t:`[expression]s`
+A :t:`type inference root` is a :t:`construct` whose inner :t:`[expression]s`
```

Comment:
Is there a benefit to generalizing this to constructs instead of just expressions? Inference deals mostly with expressions and patterns, and all of the inference roots are expressions.

Reply:
Yes, constant initializers, discriminant initializers, and static initializers have a `=` in their productions, so technically those are not expressions. This was caught by Lukas. :smile: